### PR TITLE
Boundaries API

### PR DESCRIPTION
**Pull request by vycius**
This pull request introduces the implementation of a Boundaries API using an embedded SQLite database with SpatiaLite, serving as an alternative to PostGIS in PostgreSQL. The API allows for fetching municipalities, counties, and other boundary data. 

### Key Features:
- **Standalone Docker Image**: Unlike the previous implementation (referenced in [biip-tools pull request #5](https://github.com/AplinkosMinisterija/biip-tools/pull/5)), this API operates as a standalone Docker image. It does not require a PostgreSQL database, as it utilizes an embedded SQLite database. This simplifies deployment and updating, as the latest Docker image can be pulled to apply updates.
- **SpatiaLite Integration**: The API leverages SpatiaLite for spatial data operations, making it lightweight and self-contained.
- **Small size**: Sqlite file which contains all boundaries is just 121 MB in size;

### Current Stage:
This implementation is in a proof-of-concept stage but is functional and ready for initial close testing.

### Quick testing
For quick testing just start this using docker compose:
```yaml
name: national-boundaries-api-local

services:
  national-boundaries-api:
    image: ghcr.io/govlt/national-boundaries-api:preview
    platform: linux/amd64
    pull_policy: always
    restart: unless-stopped
    ports:
      - 9321:3000
```

### Known Limitations:
- **Lack of Geometry Filtering**: The current version does not support filtering by geometry.
- **Geometry Format**: Geometry data is returned in SpatiaLite representation. To obtain WKB format, apply `HEX(AsText(geom))` in SQL queries.
- **No OpenAPI Support**: The API lacks OpenAPI documentation and support, making it challenging to auto-generate API contracts.
- **Huge docker image**: Docker image is around 2 GB;

### Framework Limitations:
- **Moleculer Framework**: The Moleculer framework has shown limitations in:
  - OpenAPI generation support.
  - Defining SQL functions within SQL select statements (only raw queries are supported).
  - Generating geometry filters (contrast with the Django Rest Framework GIS package which can automatically generate filter generation capabilities e.g. https://github.com/openwisp/django-rest-framework-gis?tab=readme-ov-file#filters).

For future iterations, it would be ideal to use a framework that:
- Provides robust OpenAPI support for auto-generating API contracts.
- Offers better support for SQL functions and geometry filtering.

Although Moleculer might be good enough for this task, it requires more manual work and more code.

### Conclusion:
This pull request lays the groundwork for a Boundaries API that is easier to deploy and update. Despite its current limitations, it is a solid starting point for further development and testing. Your feedback and contributions are welcome to enhance its functionality and address the existing shortcomings.

### CODE REVIEW

File: api/knexfile.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+const {knexSnakeCaseMappers} = require('objection');
+require('dotenv').config();
+
+export const boundariesConfig = {
+    client: 'better-sqlite3',
+    connection: {
+        filename: './boundaries.sqlite',
+    },
+    useNullAsDefault: true,
+    pool: {
```
Comment (arunas-smala) [inline api/knexfile.ts:12]: Nesu tikras dėl `pool` su `sqlite3`:

https://knexjs.org/guide/#pool
Čia mini kad `and a single connection for sqlite3 (due to issues with utilizing multiple connections on a single file)`

https://github.com/knex/knex/issues/3176
Čia minima `For now I will just follow the knex default and use config.pool.max = 1 for SQLite`

Bet nežinau, jei veikia tai gal ir ne problema,

File: api/package.json [hunk 2]
```diff
@@ -0,0 +1,93 @@
+{
+  "name": "national-boundaries-api",
+  "version": "1.0.0",
+  "description": "National boundaries API",
+  "prettier": "@aplinkosministerija/biip-prettier-config",
+  "scripts": {
+    "build": "tsc --build tsconfig.json",
+    "dev": "ts-node ./node_modules/moleculer/bin/moleculer-runner.js --env --hot --repl --config moleculer.config.ts services/**/*.service.ts",
+    "start": "moleculer-runner --config dist/moleculer.config.js",
+    "cli": "moleculer connect NATS",
```
Comment (arunas-smala) [inline api/package.json:10]: Čia mums visuose projektuose šis `cli` neaktualus, kadangi nenaudojam `NATS`

File: api/utils/index.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
+import { Readable } from 'stream';
+
+export function toReadableStream(fetchReadable: any): NodeJS.ReadableStream {
```
Comment (arunas-smala) [inline api/utils/index.ts:3]: Šis nematau, kad būtų naudojamas

**Review by arunas-smala — APPROVED** [review]
nelabai yra prie ko prikibt

**Comment by vycius** [comment]
Replaced by https://github.com/govlt/national-boundaries/pull/4

**Pull request closed without merging**